File: app/src/main/java/com/karim/booksapp/data/models/pojos/VolumeInfo.java

package com.karim.booksapp.data.models.pojos;

import java.util.List;
import com.squareup.moshi.Json;

public class VolumeInfo {

    @Json(name = "title")
    public String title;
    @Json(name = "authors")
    public List<String> authors = null;
    @Json(name = "publisher")
    public String publisher;
    @Json(name = "publishedDate")
    public String publishedDate;
    @Json(name = "description")
    public String description;
    @Json(name = "industryIdentifiers")
    public List<IndustryIdentifier> industryIdentifiers = null;
    @Json(name = "readingModes")
    public ReadingModes readingModes;
    @Json(name = "pageCount")
    public Integer pageCount;
    @Json(name = "printType")
    public String printType;
    @Json(name = "categories")
    public List<String> categories = null;
    @Json(name = "maturityRating")
    public String maturityRating;
    @Json(name = "allowAnonLogging")
    public Boolean allowAnonLogging;
    @Json(name = "contentVersion")
    public String contentVersion;
    @Json(name = "panelizationSummary")
    public PanelizationSummary panelizationSummary;
    @Json(name = "imageLinks")
    public ImageLinks imageLinks;
    @Json(name = "language")
    public String language;
    @Json(name = "previewLink")
    public String previewLink;
    @Json(name = "infoLink")
    public String infoLink;
    @Json(name = "canonicalVolumeLink")
    public String canonicalVolumeLink;
    @Json(name = "subtitle")
    public String subtitle;
    @Json(name = "averageRating")
    public Double averageRating;
    @Json(name = "ratingsCount")
    public Integer ratingsCount;

}

File: app/src/main/java/com/karim/booksapp/models/pojos/VolumeInfo.kt
package com.karim.booksapp.models.pojos

import com.squareup.moshi.Json

class VolumeInfo {

    @Json(name = "title")
    var title: String? = null
    @Json(name = "authors")
    var authors: List<String>? = null
    @Json(name = "publisher")
    var publisher: String? = null
    @Json(name = "publishedDate")
    var publishedDate: String? = null
    @Json(name = "description")
    var description: String? = null
    @Json(name = "industryIdentifiers")
    var industryIdentifiers: List<IndustryIdentifier>? = null
    @Json(name = "readingModes")
    var readingModes: ReadingModes? = null
    @Json(name = "pageCount")
    var pageCount: Int? = null
    @Json(name = "printType")
    var printType: String? = null
    @Json(name = "categories")
    var categories: List<String>? = null
    @Json(name = "maturityRating")
    var maturityRating: String? = null
    @Json(name = "allowAnonLogging")
    var allowAnonLogging: Boolean? = null
    @Json(name = "contentVersion")
    var contentVersion: String? = null
    @Json(name = "panelizationSummary")
    var panelizationSummary: PanelizationSummary? = null
    @Json(name = "imageLinks")
    var imageLinks: ImageLinks? = null
    @Json(name = "language")
    var language: String? = null
    @Json(name = "previewLink")
    var previewLink: String? = null
    @Json(name = "infoLink")
    var infoLink: String? = null
    @Json(name = "canonicalVolumeLink")
    var canonicalVolumeLink: String? = null
    @Json(name = "subtitle")
    var subtitle: String? = null
    @Json(name = "averageRating")
    var averageRating: Double? = null
    @Json(name = "ratingsCount")
    var ratingsCount: Int? = null

}
